src/mlir/generator.cc

```diff
@@ -915,6 +955,55 @@ namespace mlir::verona
     }
   }
 
+  llvm::Expected<ReturnValue> Generator::parseFieldRead(const ::ast::Ast& ast)
+  {
+    assert(AST::isMember(ast) && "Bad node");
+
+    // Find the variable to extract from
+    auto ref = AST::getLocalRef(ast);
+    auto var = symbolTable.lookup(ref);
+
+    // Get the field name
+    auto field = AST::getID(ast);
+    auto fieldAttr = StringAttr::get(field, context);
+
+    // Find the field type
+    auto type = extractClassType(var.getType());
+    auto fieldType = type.getFieldType(field);
+
+    // Return the output of the field read op
+    auto loc = getLocation(ast);
+    auto op = builder.create<FieldReadOp>(loc, fieldType, var, field);
+    return op.getResult();
+  }
+
+  llvm::Expected<ReturnValue>
+  Generator::parseFieldWrite(const ::ast::Ast& ast, mlir::Value value)
+  {
+    assert(AST::isMember(ast) && "Bad node");
+
+    // Find the variable to extract from
+    auto ref = AST::getLocalRef(ast);
+    auto var = symbolTable.lookup(ref);
+
+    // Get the field name
+    auto field = AST::getID(ast);
+    auto fieldAttr = StringAttr::get(field, context);
+
+    // Find the field type
+    auto type = extractClassType(var.getType());
```

Comment:
I think this should handle unknown. Or use unknown.

Reply:
Same answer as above.